activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb

```diff
@@ -328,7 +328,7 @@ def remove_column(name)
       #
       #   index(:account_id, name: 'index_projects_on_account_id')
       def index(column_name, options = {})
-        indexes[column_name] = options
+        indexes << options.merge(column_name: column_name)
```

Comment:
Oh, duh, totally forgot that iterating over an array of arrays will yield each item in the subarray to `each`. Much simpler change, thanks!